Remove variable length arrays in structures (VLAIS) from the netfilter code.

Signed-off-by: Mark Charlebois <charlebm@gmail.com>
---
 include/linux/jbd2.h        |    2 +-
 lib/libcrc32c.c             |    2 +-
 net/netfilter/xt_repldata.h |    6 ++++--
 3 files changed, 6 insertions(+), 4 deletions(-)

Index: linux/net/netfilter/xt_repldata.h
===================================================================
--- linux.orig/net/netfilter/xt_repldata.h	2012-09-06 17:53:50.000000000 -0400
+++ linux/net/netfilter/xt_repldata.h	2012-09-13 23:04:09.395890364 -0400
@@ -3,6 +3,8 @@
  *
  * 'entries' and 'term' are never anywhere referenced by word in code. In fact,
  * they serve as the hanging-off data accessed through repl.data[].
+ * The order or the last two elements in the struct can be changed because only
+ * the size matters since the structure members are not referenced. 
  */
 
 #define xt_alloc_initial_table(type, typ2) ({ \
@@ -11,9 +13,9 @@
 	unsigned int bytes = 0, hooknum = 0, i = 0; \
 	struct { \
 		struct type##_replace repl; \
-		struct type##_standard entries[nhooks]; \
 		struct type##_error term; \
-	} *tbl = kzalloc(sizeof(*tbl), GFP_KERNEL); \
+		struct type##_standard entries[0]; \
+	} *tbl = kzalloc(sizeof(*tbl)+nhooks*sizeof(struct type##_standard), GFP_KERNEL); \
 	if (tbl == NULL) \
 		return NULL; \
 	strncpy(tbl->repl.name, info->name, sizeof(tbl->repl.name)); \
Index: linux/lib/libcrc32c.c
===================================================================
--- linux.orig/lib/libcrc32c.c	2012-09-06 17:53:50.000000000 -0400
+++ linux/lib/libcrc32c.c	2012-09-13 23:04:09.395890364 -0400
@@ -43,7 +43,7 @@
 {
 	struct {
 		struct shash_desc shash;
-		char ctx[crypto_shash_descsize(tfm)];
+		char ctx[4];
 	} desc;
 	int err;
 
